src/profile-logic/merge-compare.js

```diff
@@ -132,21 +132,42 @@ export function mergeProfilesForDiffing(
       );
     }
     const profile = profiles[i];
-    let thread = profile.threads[selectedThreadIndex];
+    let thread = { ...profile.threads[selectedThreadIndex] };
     transformStacks[i] = profileSpecific.transforms[selectedThreadIndex];
     implementationFilters.push(profileSpecific.implementation);
 
     // We adjust the categories using the maps computed above.
     // TODO issue #2151: Also adjust subcategories.
-    thread.stackTable.category = adjustCategories(
-      thread.stackTable.category,
-      translationMapsForCategories[i]
-    );
-    thread.frameTable.category = adjustNullableCategories(
-      thread.frameTable.category,
-      translationMapsForCategories[i]
+    thread.stackTable = {
+      ...thread.stackTable,
+      category: adjustCategories(
+        thread.stackTable.category,
+        translationMapsForCategories[i]
+      ),
+    };
+    thread.frameTable = {
+      ...thread.frameTable,
+      category: adjustNullableCategories(
+        thread.frameTable.category,
+        translationMapsForCategories[i]
+      ),
+    };
+
+    const screenshotMarker = profile.threads.filter(
+      t => t.name === 'Compositor'
+    )[0].markers;
+
+    const compositorScreenshot = screenshotMarker.data.filter(
+      d => d !== null && d.type === 'CompositorScreenshot'
```

Comment:
These are just for checking, I'm planning to implement a function for the merging as @canova mentioned in the comment in the issue.